### PR TITLE
Add more examples to test logic_functions

### PR DESCRIPTION
**Pull request by shiatsumat**

### CODE REVIEW

**Comment by xldenis** [comment]
I wonder: is it better for us to have many small test files (like for traits) or fewer larger ones? What's clearer, etc?

File: creusot/tests/should_succeed/logic_functions.rs [hunk 1]
```diff
@@ -17,5 +17,22 @@ logic! {
...
+            -n + n - n * n / n
+        } else {
+            n
+        }
+    }
+}
+
+logic! {
+    fn deref_pat<'a>(p: &'a (Int, Int)) -> Int {
+        let (a, b) = p;
```
Comment (xldenis) [inline creusot/tests/should_succeed/logic_functions.rs:32]: I don't think this is testing `DerefPat`. It should say `(&a,&b)` instead.

**Review by xldenis — APPROVED** [review]
1 comment but lgtm

File: creusot/tests/should_succeed/logic_functions.rs (hunk 1 above)
Comment (shiatsumat) [reply creusot/tests/should_succeed/logic_functions.rs:35]: This is actually using `DerefPat`, turning `& (Int, Int)` into `(&Int, &Int)`.

Comment (shiatsumat) [reply creusot/tests/should_succeed/logic_functions.rs:35]: But I'll update this test to pattern-match over `enum`.

**Comment by shiatsumat** [comment]
Oh, finally it works. (I should go to bed now...)

File: creusot/tests/should_succeed/logic_functions.rs [hunk 2]
```diff
@@ -6,16 +6,35 @@
...
+        }
+    }
 }
 
-#[ensures(logical())]
-fn main() {}
+logic! {
+    fn deref_pat<'a>(o: &'a Option<Int>) -> Int {
+        match o {
+            Some(a) => *a,
```
Comment (xldenis) [inline creusot/tests/should_succeed/logic_functions.rs:36]: are you certain that this triggers the `DerefPat` branch? I really thought it was from using `&` in a pattern.

File: creusot/tests/should_succeed/logic_functions.rs (hunk 1 above)
Comment (xldenis) [reply creusot/tests/should_succeed/logic_functions.rs]: ahh ok! ignore my comments then.

**Pull request merged**